bindgen/ir/TypeDef.cpp

```diff
@@ -115,3 +127,31 @@ TypeDef::replaceType(const std::shared_ptr<const Type> &type,
     return std::make_shared<TypeDef>(
         name, this->type->replaceType(type, replacement), nullptr);
 }
+
+bool TypeDef::wrapperForOpaqueType() const { return !type; }
+
+std::string TypeDef::getCTypeName() const {
+    auto structPointer = std::dynamic_pointer_cast<const Struct>(type);
+    if (structPointer) {
+        return "struct " + structPointer->getName();
+    }
+    auto unionPointer = std::dynamic_pointer_cast<const Union>(type);
+    if (unionPointer) {
+        return "union " + unionPointer->getName();
+    }
+    auto enumPointer = std::dynamic_pointer_cast<const Enum>(type);
+    if (enumPointer) {
+        return "enum " + enumPointer->getName();
+    }
+    if (!type) {
+        if (startsWith(name, "struct_")) {
+            return "struct " +
```

Comment:
So we still encode the name with an underscore internally?

Reply:
I agree that it's better to store types with spaces